### PR TITLE
Add docstrings and doctests and fix a bug ciphers/trifid_cipher.py

### PR DESCRIPTION
**Pull request by FEROS01**
### Describe your change:

Contributes to https://github.com/TheAlgorithms/Python/issues/9943

* [x] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: ciphers/trifid_cipher.py [hunk 1]
```diff
@@ -1,8 +1,28 @@
-# https://en.wikipedia.org/wiki/Trifid_cipher
+"""
+The trifid cipher uses a table to fractionate each plaintext letter into a
+trigram,mixes the constituents of the trigrams, and then applies the table
+in reverse to turn these mixed trigrams into ciphertext letters.
+https://en.wikipedia.org/wiki/Trifid_cipher
+"""
+
 from __future__ import annotations
```
Comment (cclauss) [inline ciphers/trifid_cipher.py:10]: Let's define `character_to_number` once outside the functions.

`number_to_character` can be created from `character_to_number` in one line of code.

https://github.com/TheAlgorithms/Python/blob/579937613a6dc7e099b710e3d57767a2fab115ad/ciphers/morse_code.py#L9-L22

File: ciphers/trifid_cipher.py [hunk 2]
```diff
@@ -1,8 +1,28 @@
...
+https://en.wikipedia.org/wiki/Trifid_cipher
+"""
+
 from __future__ import annotations
 
 
 def __encrypt_part(message_part: str, character_to_number: dict[str, str]) -> str:
+    """
+    Arranges the triagram value of each letter of 'message_part' vertically
+    and joins them horizontally
```
Comment (cclauss) [inline ciphers/trifid_cipher.py:14]: Use _active_ tense like you are commanding the computer what to do.
```suggestion
    Arrange the triagram value of each letter of 'message_part' vertically and join them
    horizontally.
```

File: ciphers/trifid_cipher.py [hunk 3]
```diff
@@ -20,6 +40,20 @@ def __encrypt_part(message_part: str, character_to_number: dict[str, str]) -> st
 def __decrypt_part(
     message_part: str, character_to_number: dict[str, str]
 ) -> tuple[str, str, str]:
+    """
+    Converts each letter of the input string into there respective trigram
+    values, joins them and splits them into three equal groups of strings.
+    Then returns the group of strings .
```
Comment (cclauss) [inline ciphers/trifid_cipher.py:46]:
```suggestion
1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678
    Convert each letter of the input string into there respective trigram values, join
    them and split them into three equal groups of strings.
    Then returns the group of strings .

```

File: ciphers/trifid_cipher.py [hunk 4]
```diff
@@ -110,13 +216,35 @@ def encrypt_message(
...
+    """
     message, alphabet, character_to_number, number_to_character = __prepare(
         message, alphabet
     )
+
     decrypted_numeric = []
     decrypted = ""
 
-    for i in range(0, len(message) + 1, period):
+    for i in range(0, len(message), period):
```
Comment (cclauss) [inline ciphers/trifid_cipher.py:247]: Why this change??

File: ciphers/trifid_cipher.py (hunk 1 above)
Comment (FEROS01) [reply ciphers/trifid_cipher.py:10]: We can't hardcode the number_to_character for the functions, because the alphabet is a variable in trifid cipher. Unlike the morse code where each letter has its own representation.
But we can do it for the doctests though.

File: ciphers/trifid_cipher.py (hunk 4 above)
Comment (FEROS01) [reply ciphers/trifid_cipher.py:247]: For situations where the length of the string is a multiple of the period, the loop continues even after it had finished traversing the string, there by creating a ' list index out of range ' error in the __decrypt_part helper function. To reproduce the bug try
```python
decrypt_message('FMJFVOISSUFTFPUFEQQC','FELIXMARDSTBCGHJKNOPQUVWYZ+',5)
# Expected 'AIDETOILECIELTAIDERA'

# The length of the string is 20, a multiple of the period 5.

for i in range(0, (20+1), period)
# This will iterate 5 times [0, 5, 10, 15, 20], taking 5 letters of the string 5 times.
# The 5th time will be an empty string because 5*5 = 25 and we have just 20 letters
# Creating a 'list index out of range' error

for i in range(0, (20) , period)
# Will iterate 4 times [0, 5 10, 15], which is what we want.
# It will also work for strings with lengths that are not multiples of the period




```

**Comment by cclauss** [comment]
Just two characters... ;-)
```
ruff --output-format=github .
Error: ciphers/trifid_cipher.py:45:89: E501 Line too long (90 > 88 characters)
```

**Review by cclauss — APPROVED** [review]
Nice one!  Thanks.
